### PR TITLE
Reduce number of `tracing` and `metrics` feature gates

### PR DESCRIPTION
**Pull request by phil-opp**
Reduces the number of `#[cfg(feature = "...")]` attributes by grouping most of the the `tracing` and `metrics` code and imports into a single cfg-annotated block.

### CODE REVIEW

File: binaries/runtime/src/main.rs [hunk 1]
```diff
@@ -90,13 +83,15 @@ async fn run(
...
-
-    #[cfg(feature = "metrics")]
-    init_process_observer(meter);
+    {
+        use dora_metrics::init_meter;
+        use opentelemetry::global;
+        use opentelemetry_system_metrics::init_process_observer;
+
+        let _started = init_meter();
+        let meter = global::meter(Box::leak(node_id.to_string().into_boxed_str()));
```
Comment (haixuanTao) [inline binaries/runtime/src/main.rs:92]: The problem with putting global::meter in a conditional feature block is that it is not going to live long enough. To record metrics during the lifetime of the `dora-runtime`. I think that it is going to get dropped at the end of the block, but let me know if it's not the case.

**Review by haixuanTao — COMMENTED** [review]
It is indeed clearer with reducing the number of feature gates. Thanks for the PR.

File: binaries/runtime/src/main.rs (hunk 1 above)
Comment (phil-opp) [reply binaries/runtime/src/main.rs:92]: Ah, that could be, good catch! So `meter` should live for the whole `run` method? What about `_started`?

Comment (phil-opp) [reply binaries/runtime/src/main.rs:92]: Hmm, seems like the `init_process_observer` function takes the `meter` by value, so it is not live after the call anymore. Did you perhaps mean the `_started` binding?

**Review by haixuanTao — APPROVED** [review]
Perfect. Thanks!

**Pull request merged**